### DISCUSSION TITLE
Add properties for Redis username and password to `peppol-reporting-backend-redis`

### DISCUSSION THREAD

**Discussion by TaKO8Ki** [phax/peppol-reporting#12] [Ideas]
Hi. Thank you for working on peppol-reporting and other Peppol-related open sources! I have a plan to use peppol-reporting in phase4 and use Redis as a peppol-reporting backend. I found that `peppol-reporting-backend-redis` does not support Redis username and password currently. Can I contribute to peppol-reporting for adding the feature? If it's ok for you, I will create a pull request.

In my current plan, I will fix [`createJreisPool`](https://github.com/phax/peppol-reporting/blob/b2893531075614a968555507223ec761a53455d8/peppol-reporting-backend-redis/src/main/java/com/helper/peppol/reporting/backend/redis/PeppolReportingBackendRedisSPI.java#L79) to receive username and password like this:

Example:
```java
import redis.clients.jedis.DefaultJedisClientConfig;

public static final String CONFIG_PEPPOL_REPORTING_REDIS_USER = "peppol.reporting.redis.user";
public static final String CONFIG_PEPPOL_REPORTING_REDIS_PASSWORD = "peppol.reporting.redis.password";

@Nullable
@OverrideOnDemand
protected JedisPool createJedisPool (@Nonnull final IConfig aConfig)
{
  final String sHost = aConfig.getAsString (CONFIG_PEPPOL_REPORTING_REDIS_HOST);
  if (StringHelper.hasNoText (sHost))
  {
    LOGGER.error ("The Redis hostname is missing in the configuration. See property '" +
                  CONFIG_PEPPOL_REPORTING_REDIS_HOST +
                  "'");
    return null;
  }

  final int nPort = aConfig.getAsInt (CONFIG_PEPPOL_REPORTING_REDIS_PORT, DEFAULT_REDIS_PORT);
  if (nPort < 0)
  {
    LOGGER.error ("The Redis port name is missing in the configuration. See property '" +
                  CONFIG_PEPPOL_REPORTING_REDIS_PORT +
                  "'");
    return null;
  }

  // =========== changes ===========
  final String username = aConfig.getAsString (CONFIG_PEPPOL_REPORTING_REDIS_USER);
  final String password = aConfig.getAsString (CONFIG_PEPPOL_REPORTING_REDIS_PASSWORD);

  LOGGER.info ("Using Peppol Reporting Redis at '" + sHost + ":" + nPort + "'");
  if (StringHelper.hasText (username) && StringHelper.hasText (password))
  {
    return new JedisPool (sHost, nPort, username, password);
  } else {
    return new JedisPool (sHost, nPort);
  }
  // ==============================
}
```

**Comment by phax** [comment]
Hello @TaKO8Ki ,
That is a wonderful idea - just go ahead.
This is the first time I was dealing with Redis, so my knowledge is a bit limited.
Just go ahead and create a PR - I am happy to include it. And don't bother about the formatting - I will clean it up later.
Thanks, Philip

**Reply by TaKO8Ki** [reply]
Thanks! I have created a pull request https://github.com/phax/peppol-reporting/pull/13.

**Reply by phax** [reply]
Thanks - approved and part of the 2.2.3 release

**Reply by TaKO8Ki** [reply]
Thanks!
